homeassistant/components/wolflink/sensor.py

```diff
@@ -0,0 +1,174 @@
+    """Set up all entries for Wolf Platform."""
+
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]["coordinator"]
+    parameters = hass.data[DOMAIN][config_entry.entry_id]["parameters"]
+
+    entities = []
+    for parameter in parameters:
+        if isinstance(parameter, Temperature):
+            entities.append(WolfLinkTemperature(coordinator, parameter))
+        if isinstance(parameter, Pressure):
+            entities.append(WolfLinkPressure(coordinator, parameter))
+        if isinstance(parameter, PercentageParameter):
+            entities.append(WolfLinkPercentage(coordinator, parameter))
+        if isinstance(parameter, ListItemParameter):
+            entities.append(WolfLinkState(coordinator, parameter))
+        if isinstance(parameter, HoursParameter):
+            entities.append(WolfLinkHours(coordinator, parameter))
+        if isinstance(parameter, SimpleParameter):
+            entities.append(WolfLinkSensor(coordinator, parameter))
+
+    async_add_entities(entities, True)
+
+
+class WolfLinkSensor(Entity):
+    """Base class for all Wolf entities."""
+
+    def __init__(self, coordinator, wolf_object: Parameter):
+        """Initialize."""
+        self.coordinator = coordinator
+        self.wolf_object = wolf_object
+
+    @property
+    def name(self):
+        """Return the name."""
+        return f"{self.wolf_object.name}"
+
+    @property
+    def state(self):
+        """Return the state."""
+        return self.coordinator.data[self.wolf_object.value_id]
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            "parameter_id": self.wolf_object.parameter_id,
+            "value_id": self.wolf_object.value_id,
+            "parent": self.wolf_object.parent,
+        }
+
+    @property
+    def unique_id(self):
+        """Return a unique_id for this entity."""
+        return self.wolf_object.parameter_id
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return bool(self.state)
+
+    @property
+    def should_poll(self):
+        """No need to poll. Coordinator notifies entity of updates."""
+        return False
+
+    async def async_added_to_hass(self):
+        """When entity is added to hass."""
+        self.async_on_remove(
+            self.coordinator.async_add_listener(self.async_write_ha_state)
+        )
+
+    async def async_update(self):
+        """Update the sensor."""
+        await self.coordinator.async_request_refresh()
+        _LOGGER.debug("Updating %s", self.coordinator.data[self.wolf_object.value_id])
+
+
+class WolfLinkHours(WolfLinkSensor):
+    """Class for hour based entities."""
+
+    @property
+    def icon(self):
+        """Icon to display in the front end."""
+        return "mdi:clock"
+
+    @property
+    def device_class(self):
+        """Return the device_class."""
+        return DEVICE_CLASS_TIMESTAMP
```

Comment:
The timestamp device class sensor should return an absolute isoformatted UTC datetime timestamp as state.

Reply:
I cannot return UTC datetime timestamp here. Or maybe I am not aware of that. Those variables are returning variable like: "How many hours heating system have been used", so it is like hours counter not timestamp. Maybe different sensor type is prepared for such usage.

Reply:
We could calculate an absolute timestamp from now, but it sounds like this measurement makes more sense as just hours counted instead of a timestamp.

Reply:
So should I leave this DEVICE_CLASS_TIMESTAMP or get rid of it? If this class stays then I will remove mdi:clock icon.

Reply:
I suggest we remove device class and set unit of measurement to `TIME_HOURS`.

https://github.com/home-assistant/core/blob/93c6a9cd96cad036a465210f41d6d377d3e284e5/homeassistant/const.py#L385

Reply:
With mdi:clock I guess, right?